### DISCUSSION TITLE
Request: open source, documentation, contact

### DISCUSSION THREAD

**Discussion by braco** [shinsenter/php#37] [Ideas]
Hi @shinsenter!

I have a request for:

https://hub.docker.com/r/shinsenter/flarum

Can you
1. Provide a repo for the source of this, is that here? https://github.com/shinsenter/php/blob/ea74944ac3f6dd5757c182b79de5af4cfb8c62e8/src/webapps/flarum/README.md?plain=1
2. Provide an example for adding a "composer require" command in the Dockerfile (I'm not sure how to do this without causing a problem with the entrypoint
3. Clearer way to open issues or contact, do you want flarum issues posted on your /php repo, for example?

Thank you :)

**Comment by shinsenter** [comment]
Hi @braco,

1. Yes, the source code of the Dockerfile and a start-up script for the flarum docker images are located [here](https://github.com/shinsenter/php/tree/main/src/webapps/flarum). Please feel free to download files in that directory and customize for your project as needed.
2. You may modify the [add-flarum](https://github.com/shinsenter/php/blob/main/src/webapps/flarum/root/startup/add-flarum) script by adding your preferred compose commands.
3. I would appreciate if any suggestions or discussions regarding my docker images could be posted in this repository. As this is my sole repository for all docker images published to Docker Hub, and I am currently not receiving an overwhelming volume of inquiries on any singular image, consolidation in one place should suffice. I do not have any specific templates to follow - please share your thoughts in whichever format you find most suitable. I will make every effort to address all discussions to the best of my abilities.

Please let me know if you need any clarification or have additional questions!

**Comment by braco** [comment]
Thanks for the replies, I'll check out those options.

It might be interesting to have an optional ENV like `SCRIPT_AFTER_BUILD=./file.sh` to hook into that part of the build process without having to duplicate existing scripts. This way the user's Dockerfile can build off of yours and still receive updates.

A link to this github repo in the Docker readme would be helpful – I had to google around a bit to find this repo. 

Might be some ideas in here: https://github.com/mondediefr/docker-flarum

Thanks again!

**Comment by braco** [comment]
Oh it looks like you might have a convention for startup scripts, with root/startup and

`ADD root/ /`

maybe an `after-build` folder with scripts?

**Comment by shinsenter** [comment]
Hi @braco 

If you download and edit the files in the [flarum](https://github.com/shinsenter/php/tree/main/src/webapps/flarum) directory I mentioned above, there will be no duplicate files at all, because docker will build a completely new image for flarum from my base image `shinsenter/phpfpm-apache`.

---

I also don't think it's necessary to add an `SCRIPT_AFTER_BUILD` environment variable as you suggested, because when you want to run some script after the docker image building process, just add your command lines to the `Dockerfile`, it will be the most optimal way.
E.g.

```dockerfile
FROM shinsenter/flarum:8.2

# your custom commands will be here
ADD file.sh /tmp/file.sh
RUN sh /tmp/file.sh
```

I am not sure if I understand your intention correctly, if you want to run some scripts RIGHT AFTER STARTING the docker container using my base image, you can copy your script to the `/startup` directory of the container, similar to the [add-flarum](https://github.com/shinsenter/php/blob/main/src/webapps/flarum/root/startup/add-flarum) script in the example I made. 
E.g.

```dockerfile
FROM shinsenter/flarum:8.2

# copy your startup scripts to the /startup directory
COPY file.sh /startup/file.sh

```

---

Lastly, about the documentation, I think my documents may be hard to understand for some people. So it would be very grateful if you could help me improve these documents by creating a pull request. I will review and merge your pull request if it is more understandable.

Best regards.

**Reply by braco** [reply]
Thank you @shinsenter – I don't think either of those techniques work for composer commands, because the order of the script needs to be guaranteed to run synchronously and only after add-flarum is complete:

```
cd $(webhome)
composer ...
```

If there's some ordinal logic to startup scripts, maybe it would be good to prefix the files with a number, like "10-install-thing"

In the meantime it seems like you need to copy the add-flarum script in its entirety and add lines at the bottom of the copied script.

I also needed to chmod in the Dockerfile:

```
ADD root/ /
RUN chmod a+x /startup/*
```

**Reply by shinsenter** [reply]
@braco 

> maybe it would be good to prefix the files with a number, like "10-install-thing"

Yes, you are right. The scripts in the startup directory will be run in alphabetical order, so you can absolutely add numbers, or add prefixes like 'zzz-my-script.sh' to decide the order of scripts you want to execute.

> I also needed to chmod in the Dockerfile

For me, it is unnecessary to run the chmod command inside Dockerfile because my script has already been granted execute permissions on the host machine. I think this is optional depending on each person's approach.

**Reply by shinsenter** [reply]
@braco 

We are rolling out updated docker images (v5.x) based on the official php images, ensuring packages receive the latest bug fixes from upstream compared to the old image versions.
https://hub.docker.com/r/shinsenter/php/

We are using different docker image tags to maintain the old docker image versions. All old docker images have been published with *-deprecated tag names. However, we will no longer provide any update for the old docker images.
https://hub.docker.com/r/shinsenter/php/tags?page=1&name=-deprecated

<img width="807" alt="Screenshot 2024-02-02 at 8 35 59" src="https://github.com/shinsenter/php/assets/2082119/e4dcee17-55d1-4ac7-9d08-5315eb9f63d7">

Please let me know if you have any other questions or concerns - we're committed to making this a seamless process.

Best regards.
